Clarify that exists? and find look in the table, not the unsaved collection (#40129)
rails/rails: guides/source/association_basics.md

##### `collection.find(...)`

> The `collection.find` method finds objects within the collection by querying the table. It uses the same syntax and options as

Reviewer: ```suggestion
The `collection.find` method finds objects within the collection's table. It uses the same syntax and options as
```

Author: I like this too. Done

> The `collection.find` method finds objects within the table (not an unsaved collection). It uses the same syntax and options as

Reviewer 2: ```suggestion
The `collection.find` method finds objects within the collection by querying the database. It uses the same syntax and options as
```

> not an unsaved collection

The collection being saved or unsaved  is not relevant, so we can drop that out.

Author: I changed it to your wording but added explicitly "It will not find  objects that are not in the table" because "querying the table" does not necessarily imply _not_ finding items in the collection but not in the table, and being explicit is useful anyway.

##### `collection.exists?(...)`

> The `collection.exists?` method checks whether an object meeting the supplied
> conditions exists in the collection. It uses the same syntax and options as

Reviewer: We're losing the note about being able to supply a condition to `exists?` with the change here so the new documentation doesn't read better to me. Perhaps the following change is sufficient to clear up the doubts here? 

```suggestion
The `collection.exists?` method checks whether an object within the collection by querying the table. It uses the same syntax and options as
```

##### `collection.exists?(...)`

> The `collection.exists?` method checks whether an object within the collection by querying the table. It will not find

Reviewer 2: > It will not find objects that are not in the table.

Let's remove these, its redundant information.

Author: Done

Reviewer 3: >It will not find objects that are not in the table.

This sentence is still there.

##### `collection.exists?(...)`

The `collection.exists?` method checks whether an object meeting the supplied

> conditions exists in the table (not an unsaved collection). It uses the same syntax and options as

Reviewer 2: Same as above

Author: Same as above
